packages/react/src/components/Grid/Grid.js

```diff
@@ -0,0 +1,311 @@
+  children: PropTypes.node,
+
+  /** Pass through any other undefined props to DOM */
+  rest: PropTypes.node,
+};
+
+/**
+ * // helper typedefs to make raw jsdoc easier to read
+ * @typedef {number | boolean | "auto"} span non-object version of breakpoint def
+ * @typedef {span | { span: span, offset: number }} breakPtRaw complete (wide) def according to prop
+ * @typedef {{ span: number | "-auto", offset: number }} breakPt processed to common obj format
+ */
+
+/**
+ * process wide variety of accepted values of Grid.Col breakpoint props (sm, lg, etc.)
+ * into standard obj format to greatly simplify syntax of Grid.Col component definition
+ *
+ * @param {breakPtRaw} breakpoint - raw def as allowed by Grid.Col breakpoint props
+ * @returns {breakPt} processed to common obj format
+ */
+const standardize = breakpoint => {
+  const obj =
+    typeof breakpoint !== 'object' ? { span: breakpoint } : breakpoint;
+
+  // vastly improves component readability by converting to format of CSS classNames
+  if (obj.span === 'auto' || obj.span === true) obj.span = '-auto';
+
+  return obj;
+};
+
+export const GridCol = ({
+  as = 'div',
+  sm: smRaw,
+  md: mdRaw,
+  lg: lgRaw,
+  xlg: xlgRaw,
+  max: maxRaw,
+  noGutter = false,
+  className = '',
+  children = null,
+  ...rest
+}) => {
+  // map list of raw breakpoints into obj format to simplify below className definition
+  // extra arr created and destructured since same transformation's applied to each bp
+  const [sm, md, lg, xlg, max] = [smRaw, mdRaw, lgRaw, xlgRaw, maxRaw].map(
+    breakPt => standardize(breakPt)
+  );
+
+  return React.createElement(
+    as,
+    {
+      ...rest,
+      className: classNames(
+        `${prefix}--col`,
+        sm.span && `${prefix}--col-sm-${sm.span}`,
+        md.span && `${prefix}--col-md-${md.span}`,
+        lg.span && `${prefix}--col-lg-${lg.span}`,
+        xlg.span && `${prefix}--col-xlg-${xlg.span}`,
+        max.span && `${prefix}--col-max-${max.span}`,
+
+        sm.offset && `${prefix}--offset-sm-${sm.offset}`,
+        md.offset && `${prefix}--offset-md-${md.offset}`,
+        lg.offset && `${prefix}--offset-lg-${lg.offset}`,
+        xlg.offset && `${prefix}--offset-xlg-${xlg.offset}`,
+        max.offset && `${prefix}--offset-max-${max.offset}`,
+
+        noGutter === true && `${prefix}--no-gutter`, // true => both
+        noGutter && noGutter !== true && `${prefix}--no-gutter--${noGutter}`, // "left" || "right"
+        className
+      ),
+    },
+    children
+  );
+};
+
+GridCol.propTypes = {
+  /**
+   * Provide a custom element to render instead of the default <div>
+   */
+  as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+
+  /**
+   * Specify column span at this width.
+   *
+   * Only supports width of 4 total columns.
+   *
+   * Screen size >= 320px
+   *
+   * @see https://www.carbondesignsystem.com/guidelines/layout#breakpoints
+   */
+  sm: PropTypes.oneOfType([
+    PropTypes.oneOf(VALID_COL_WIDTHS.sm),
```

Comment:
Since some folks might be using these components in different grid modes (12 or 16) would it make sense to loosen this prop type to `PropTypes.number`?

Reply:
If I'm not mistaken, since `VALID_COL_WIDTHS` is generated from the `@carbon/layout` package, wouldn't they just be the right value of up to 12 or 16 depending on the user's settings?